e2e/utils/ProfileUtils.ts

```diff
@@ -0,0 +1,370 @@
+  const fullPath = path.join(paths.config, 'settings.json');
+
+  try {
+    await fs.promises.access(fullPath);
+  } catch {
+    createDefaultSettings();
+  }
+
+  return [];
+}
+
+export async function verifyNoRegistryHive(hive: string): Promise<string[]> {
+  const skipReasons: string[] = [];
+
+  for (const profileType of ['defaults', 'locked']) {
+    try {
+      const { stdout } = await childProcess.spawnFile('reg',
+        ['query', `${ hive }\\SOFTWARE\\Policies\\Rancher Desktop\\${ profileType }`],
+        { stdio: ['ignore', 'pipe', 'pipe'] });
+
+      if (stdout.length === 0) {
+        continue;
+      }
+    } catch { }
+    skipReasons.push(`Need to remove registry hive "${ hive }\\SOFTWARE\\Policies\\Rancher Desktop\\${ profileType }"`);
+  }
+
+  return skipReasons;
+}
+
+export async function verifyUserProfile(): Promise<string[]> {
+  const platform = os.platform() as 'win32' | 'darwin' | 'linux';
+  // console.log('check settings');
+  // await util.promisify(setTimeout)(2 * 60_000);
+
+  if (platform === 'win32') {
+    return verifyRegistryHive('HKCU');
+  }
+  for (const profilePath of getDeploymentPaths(platform, paths.deploymentProfileUser)) {
+    try {
+      await fs.promises.access(profilePath);
+
+      return [];
+    } catch { }
+  }
+  await createUserProfile(
+    { containerEngine: { allowedImages: { enabled: true } } },
+    { containerEngine: { allowedImages: { enabled: true, patterns: [__filename] } }, kubernetes: { version: 'chaff' } },
+  );
+
+  return [];
+}
+
+export async function verifyNoSystemProfile(): Promise<string[]> {
+  const platform = os.platform() as 'win32' | 'darwin' | 'linux';
+
+  if (platform === 'win32') {
+    return await verifyNoRegistryHive('HKLM');
+  }
+  const profilePaths = getDeploymentPaths(platform, paths.deploymentProfileSystem);
+  const existingProfiles = [];
+
+  for (const profilePath of profilePaths) {
+    if (await fileExists(profilePath)) {
+      existingProfiles.push(`Need to delete system profile ${ profilePath }`);
+    }
+  }
+
+  return existingProfiles;
+}
+
+export async function verifySystemProfile(): Promise<string[]> {
+  const platform = os.platform() as 'win32' | 'darwin' | 'linux';
+  // console.log('check settings');
+  // await util.promisify(setTimeout)(2 * 60_000);
+
+  if (platform === 'win32') {
+    return await verifyRegistryHive('HKLM');
+  }
+  const profilePaths = getDeploymentPaths(platform, paths.deploymentProfileSystem);
+
+  for (const profilePath of profilePaths) {
+    if (await fileExists(profilePath)) {
+      return [];
+    }
+  }
+
+  return [`Need to create system profile file ${ profilePaths.join(' and/or ') }`];
+}
+
+// And the test runners.
+// There are only three kinds of tests, so each of the main test files can invoke the kind it needs:
+// 1. Verify there's a first-run window
+// 2. Verify the main window is the first window
+// 3. Verify we get a fatal error and it's captured in a log file.
```

Comment:
I added pointers to the main block -- it makes more sense to read the explanation of all three kinds of functions.

Reply:
I see there is a space between the pointer comments and the function that they refer to - is this intentional? I hate to bikeshed but this seems weird.

Reply:
Reasonable question. I did that deliberately because that block of comments refers to the following three functions, not just the next one, so it felt reasonable to separate it from the following function.